runtime/src/main/java/com/redhat/ipaas/rest/ComponentGroups.java

```diff
@@ -37,12 +38,15 @@
     @Inject
     private DataManager dataMgr;
 
+    @Context
+    private UriInfo uri;
+
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "List component groups")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "Success", response = ComponentGroup.class)})
     public Collection<ComponentGroup> list() {
-        return dataMgr.fetchAll(ComponentGroup.class);
+        return dataMgr.fetchAll(ComponentGroup.class, new ReflectiveSorter<>(ComponentGroup.class, uri.getQueryParameters()));
```

Comment:
Yes, we could. But I thought it would better to let the sorter pick the properties he wants to, so that the endpoints don't kneed the knowledge. Also in case of a change or maybe an additional parrameter to sort (e.g. extracting multiple fields for a secondary sort order), one would only need to change the sorter. 

The drawback of course is that it couples the ReflectiveSorter to jax-rs' MultiValueMap so that it cant be reused in a different context.